ipld/add.go

```diff
@@ -35,6 +35,28 @@ func AddShares(ctx context.Context, shares []Share, adder ipld.NodeAdder) (*rsmt
 	return eds, batchAdder.Commit()
 }
 
+// ImportShares imports flattend chunks of data into Extended Data square and saves it in IPLD DAG
+func ImportShares(ctx context.Context, shares [][]byte, na ipld.NodeAdder) (*rsmt2d.ExtendedDataSquare, error) {
+	if len(shares) == 0 {
+		return nil, fmt.Errorf("empty data")
```

Comment:
```suggestion
		return nil, fmt.Errorf("ipld: importing empty data")
```

Reply:
Alternatively, I can suggest `Converter` as it converts shares from` core.Block` to `ExtendedHeader`

Reply:
Converting is about transforming something from one form into another. Here we convert multiple inputs into one, which is not really converting, and the resulting Header is not a cumulative of all the inputs, as it does not store has shares which are passed to the func in the form of a block. I would propose `header.Commiter` instead. It is not that broad as `Maker` and semantically close to blockchain/crypto terminology.

Reply:
cc @renaynay

Reply:
I am in favour of either 

* `Generator`
* `Constructor` 

Either is fine with me bc encompasses the general thing that's happening within the function.

Reply:
The function does much more stuff to be called `Constructor`, but `Generator` is fine. I would still be more precise with `Commiter` as it also involves what `Generator` means in some way and also makes it more connected to what we are doing. Basically, Bridge Node that executes this function commits to the correctness of the Header and distributes it on the network so others can verify the commitment/header. 

cc @liamsi, WDYT?

Reply:
`Commiter` and `Generator` are both ok for me

Reply:
`Generator`, lets just ship. I am not in favour of `Committer`. If we both like `Generator`, lets just compromise and use.

Reply:
Changed to generator

Reply:
I'd even call it sth like `header.ConstructExtendedFn` or `header.CreateExtendedFn`(even the comment above the type definition says "a function that creates an ExtendedHeader" so the latte would call the type accordingly). You can skip the `Fn` suffix.


Just regarding the nuances of the verbs used here: 
generate/create is to bring something into being
construct is to build or form (something) by assembling parts

Reply:
Hlib mentioned to me that this is planned to become an interface. then the name should follow golang naming conventions for interfaces: 
> By convention, one-method interfaces are named by the method name plus an -er

IMO, we can do this once this will be an interface (and not necessarily in this PR).

Reply:
I personally like `Fn` suffix. It's descriptive and good 👍🏻 

I'm in favour of `header.ConstructExtendedFn`.

Reply:
`WithHeaderConstructorFn` but let's get @Wondertan approval first

Reply:
Ok, let's do `header.ConstructFn`

Reply:
renamed